packages/samples/headless-react/src/components/triggers/query-trigger.class.tsx

```diff
@@ -0,0 +1,34 @@
+import {Component, ContextType} from 'react';
+import {
+  buildQueryTrigger,
+  QueryTrigger as HeadlessQueryTrigger,
+  QueryTriggerState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {AppContext} from '../../context/engine';
+
+export class QueryTrigger extends Component<{}, QueryTriggerState> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
+
+  private controller!: HeadlessQueryTrigger;
+  private unsubscribe: Unsubscribe = () => {};
+
+  componentDidMount() {
+    this.controller = buildQueryTrigger(this.context.engine!);
+    this.updateState();
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  render() {
+    return null;
```

Comment:
For the samples, it would be nice to return a message explaining that the query changed from `a` to `b`.